src/vision/Vision.cpp

```diff
@@ -37,104 +65,102 @@ void leftCamCallback(const wfov_camera_msgs::WFOVImage::ConstPtr& msg)
+                    circle(original, center, 4, Scalar(0, 0, 255), -1);
+                }
             }
-        }
-
-        //Find the moments (physical properties) of the contour
-        Moments moment;
-        moment = moments(contours[largestIndex], false);
 
-        int cx = -1;
-        int cy = -1;
-        int imWidth = msg->image.width;
-        int imHeight = msg->image.height;
-        //Using moments, find the center point of the contour
-        if (moment.m00 > 0)
-        {
-            //Find x and y coordinates with 0,0 being top left corner
-            cx = static_cast<int>(moment.m10/moment.m00);
-            cy = static_cast<int>(moment.m01/moment.m00);
-            Point2f center = cv::Point2f(cx, cy);
-            std::cout << "Center at: " << "[" << cx - (imWidth/2) << "," <<
-                                    -1*(cy-(imHeight / 2)) << "]" << std::endl;
-            circle(original, center, 5, Scalar(255, 255, 255), -1);
-            //Draw a circle on the original image for location visualization
-            circle(original, center, 4, Scalar(0, 0, 255), -1);
+            ROS_DEBUG_STREAM("Preparing output");
+            ///Create the output message
+            robosub::visionPos outMsg;
+
+            //Prepare the output message
+            outMsg.xPos = cx - (imWidth / 2);
+            outMsg.yPos = cy - (imHeight / 2);
+            outMsg.magnitude = static_cast<double>(contourArea(contours[i],
+                                                   false)) /
+                               static_cast<double>(imWidth * imHeight);
+            //Add to output
+            arrayOut.data.push_back(outMsg);
         }
-
-        ROS_DEBUG_STREAM("Preparing output");
-        //Prepare the output message
-        outMsg.xPos = cx - (imWidth / 2);
-        ROS_DEBUG_STREAM("X prepared");
-        outMsg.yPos = cy - (imHeight / 2);
-        ROS_DEBUG_STREAM("Y prepared");
-        outMsg.magnitude = static_cast<double>(largestArea) /
-                           static_cast<double>(imWidth * imHeight);
-        ROS_DEBUG_STREAM("Magnitude prepared");
     }
 
     //Show images
-    namedWindow("Original");
-    imshow("Original", original);
-
-    namedWindow("left_mask");
-    imshow("left_mask", procOut);
+    if (doImShow)
+    {
+        namedWindow(ros::this_node::getName() + " Original");
+        imshow(ros::this_node::getName() + " Original", original);
 
-    //Wait for 1 millisecond to show images
-    waitKey(1);
+        namedWindow(ros::this_node::getName() + " left_mask");
+        imshow(ros::this_node::getName() + " left_mask", procOut);
+        //Wait for 1 millisecond to show images
+        waitKey(1);
+    }
 
     //Publish output message
-    ROS_DEBUG_STREAM("Publishing message");
-
-    robosub::visionPosArray arrayOut;
-    arrayOut.data.push_back(outMsg);
     pub.publish(arrayOut);
 }
 
 void rightCamCallback(const wfov_camera_msgs::WFOVImage::ConstPtr& msg)
```

Comment:
Though it does not currently serve any function, we will in future want to take advantage of the right camera as well. For now, I have commented out the related code and changed it from using a WFOVImage, though it will be needed again later.